### PR TITLE
Implement export rl matrix

### PR DESCRIPTION
**Pull request by gmalinve**
PR to export RL matrix

### CODE REVIEW

File: pyaedt/application/Analysis.py [hunk 1]
```diff
@@ -1849,3 +1850,104 @@ def value_with_units(self, value, units=None):
...
+            if matrix_list:
+                if not [
+                    matrix
+                    for matrix in matrix_list
+                    if matrix.name == matrix_name or [x for x in matrix.available_properties if matrix_name in x]
+                ]:
+                    msg = "Matrix name doesn't exist, provide and existing matrix name."
+                    raise ValueError(msg)
+            else:
+                msg = "Matrix list parameters is empty, can't export a valid matrix."
```
Comment (Samuelopez-ansys) [inline pyaedt/application/Analysis.py:1914]: @gmalinve 

You should modify the raise error with:

_self.logger.error(msg)
return False_

If you see any other pyaedt method, it returns False when it fails with a logger error message.

And in the unittest you could also add these tests, I mean, test when it fails (when filepath is None or the extension is not .txt) this will improve even more the coverage.

Comment (Samuelopez-ansys) [inline pyaedt/application/Analysis.py:1914]: This should return False, right?

**Review by Samuelopez-ansys — CHANGES_REQUESTED** [review]

File: pyaedt/application/Analysis.py [hunk 2]
```diff
@@ -1849,3 +1850,104 @@ def value_with_units(self, value, units=None):
                 units = self.modeler.model_units
             val = "{0}{1}".format(value, units)
         return val
+
+    # Add test and check if Analysis.py is the best place to have the method
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:1854]:
```suggestion
```

File: pyaedt/application/Analysis.py [hunk 3]
```diff
@@ -1849,3 +1850,104 @@ def value_with_units(self, value, units=None):
...
+            Matrix name to be exported.
+        file_path : str
+            File path to export R/L matrix file.
+        is_format_default : bool, optional
+            Whether the exported format is default or not.
+            If False the custom format is set (no exponential).
+        width : int, optional
+            Column width in exported .txt file.
+        precision : int, optional
+            Decimal precision number in exported .txt file.
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:1882]:
```suggestion
            Decimal precision number in exported *.txt file.
```

File: pyaedt/application/Analysis.py [hunk 4]
```diff
@@ -1849,3 +1850,104 @@ def value_with_units(self, value, units=None):
...
+        is_exponential : bool, optional
+            Whether the format number is exponential or not.
+        setup_name : str, optional
+            Name of the setup.
+        default_adaptive : str, optional
+            Adaptive type.
+        is_post_processed : bool, optional
+            Boolean to check if it is post processed.
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:1890]:
```suggestion
            Boolean to check if it is post processed. Default value is ``False``.
```

File: pyaedt/application/Analysis.py [hunk 5]
```diff
@@ -1849,3 +1850,104 @@ def value_with_units(self, value, units=None):
...
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+        """
+        try:
+            if not self.solution_type == "EddyCurrent":
+                msg = "RL Matrix can only be exported if solution type is Eddy Current."
+                raise ValueError(msg)
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:1900]: Usually, it is not needed to create a new variable when it is used only one time.
```suggestion
                raise ValueError("RL Matrix can only be exported if solution type is Eddy Current.")
```

**Review by Samuelopez-ansys — APPROVED** [review]
lgtm
